--- boost/atomic/atomic.hpp	2012-12-24 22:36:31.000000000 -0500
+++ boost/atomic/atomic.hpp	2014-05-06 06:13:34.625140562 -0400
@@ -158,7 +158,7 @@
 #if !defined(__PGIC__)
 
 #if (defined(BOOST_WINDOWS) && !defined(_WIN32_WCE)) \
-    || (defined(_XOPEN_UNIX) && (_XOPEN_UNIX+0 > 0)) \
+    || (defined(_XOPEN_UNIX) && (_XOPEN_UNIX+0 > 0) && !defined(__UCLIBC__) && !defined(__GLIBC__)) \
     || defined(__CYGWIN__) \
     || defined(macintosh) || defined(__APPLE__) || defined(__APPLE_CC__) \
     || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
